src/Extensions/Diagnostics.HealthChecks/Internal/HealthCheckPublisher.cs

```diff
@@ -3,55 +3,41 @@
 
 using System;
 using System.Collections.Generic;
-using System.Fabric;
 using System.Text;
 using System.Threading;
 using System.Threading.Tasks;
 using Microsoft.Extensions.Diagnostics.HealthChecks;
 using Microsoft.Extensions.ObjectPool;
-using ServiceFabricHealth = System.Fabric.Health;
 
 namespace Microsoft.Omex.Extensions.Diagnostics.HealthChecks
 {
 	internal abstract class HealthCheckPublisher : IHealthCheckPublisher
 	{
-		protected readonly ObjectPool<StringBuilder> StringBuilderPool;
+		internal readonly ObjectPool<StringBuilder> StringBuilderPool;
 
 		internal abstract string HealthReportSourceId { get; }
 
-		internal const string HealthReportSummaryProperty = "HealthReportSummary";
+		internal const string HealthReportSummaryProperty = "HealthReportSummary";
 
-		public HealthCheckPublisher(ObjectPoolProvider objectPoolProvider) => StringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
+		public HealthCheckPublisher(ObjectPoolProvider objectPoolProvider) => StringBuilderPool = objectPoolProvider.CreateStringBuilderPool();
 
-		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
+		public abstract Task PublishAsync(HealthReport report, CancellationToken cancellationToken);
 
-		private ServiceFabricHealth.HealthInformation FinalizeHealthReport(string healthReportSummaryProperty, ServiceFabricHealth.HealthState healthState)
-		{
-			return new ServiceFabricHealth.HealthInformation(HealthReportSourceId, healthReportSummaryProperty, healthState);
-		}
-		protected void PublishAllEntries(HealthReport report, Action<ServiceFabricHealth.HealthInformation> publishFunc,
+		protected void PublishAllEntries(HealthReport report, Func<string, HealthStatus, string, Task> publishFunc,
```

Comment:
I'm not sure that using Func here is a good approach, parameter list like this `string, HealthStatus, string, Task` very confusing. Could we make it an abstract function? Or extract functionality into the interface.